[Lavanya Mishra] iP

Lava-Iris opened the pull request:
## Iris
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Iris frees your mind of having to remember things you need to do. It is:
- text based
- easy to learn 
- ~~Fast~~ _SUPER_ FAST to use

All you need to do is:
1. download it from [here](https://github.com/Lava-Iris/ip/releases/tag/A-Jar)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Reminders (coming soon)

# 
If you are a Java programmer, you can use it to practice Java too. Here is the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Lava-Iris committed:
Level-1 Baby chatbot

Lava-Iris committed:
Level-2 Teenage storage chatbot

Lava-Iris committed:
changed the greeting statement

Lava-Iris committed:
added task managmement functionality

Lava-Iris committed:
added support for three types of tasks: todos, events and deadlines

Lava-Iris committed:
added Automatic i/o testing

Lava-Iris committed:
added support for deleting tasks from task list

Lava-Iris committed:
Using enums for readability

Lava-Iris committed:
handled errors such as missing tasks, missing fields, unknown commands, and wrong fields

Lava-Iris committed:
added delete as a command in help

Lava-Iris committed:
added new commands in test input to test new features

Lava-Iris committed:
change the names of fields and classes

Lava-Iris committed:
improve code quality

Lava-Iris committed:
store tasks in hard drive

Lava-Iris committed:
treat date and time as date instead of string

Lava-Iris committed:
Merge branch 'branch-Level-8'

Lava-Iris committed:
see date and time as date instead of string

Lava-Iris committed:
remove minor errors

Lava-Iris committed:
add command to filter tasks based on date-time

Lava-Iris committed:
organise the classes into packages

Lava-Iris committed:
organise the classes into packages

Lava-Iris committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Lava-Iris committed:
Set up gradle

Lava-Iris committed:
add jUnit tests for task classes

Lava-Iris committed:
change duke to iris in some places

Lava-Iris committed:
tweak the code to follow given coding standard

Lava-Iris committed:
add javadoc to all the public classes and methods

Lava-Iris committed:
allow users to find a task by searching a keyword

Lava-Iris committed:
resolved merge conflicts

Lava-Iris committed:
resolved merge conflicts

Merrickneo commented on src/main/java/iris/TaskStore.java:
```diff
@@ -0,0 +1,98 @@
+        TaskList tasks = new TaskList();
+        try {
+            try {
+                Scanner fileScanner = new Scanner(this.taskFile);
```
Remember to close your scanner object!

Lava-Iris replied:
OH! I completely forgot about it. Thank you for the reminder! :)

Merrickneo commented on src/main/java/iris/task/Event.java:
```diff
@@ -0,0 +1,60 @@
+    public Event(String name, String from, String to) throws DateTimeException {
+        super(name);
+        try {
+            this.from = LocalDateTime.parse(from.trim(), DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm"));
```
Just a suggestion but you could perhaps create another class like `DatetimeParser` to help with parsing of the datetime string!

Lava-Iris replied:
Oh I was thinking of making a method for that (it's kinda messy and repetitive now), but I think a class might be better. Thank you!

Merrickneo commented on src/main/java/iris/Parser.java:
```diff
@@ -0,0 +1,134 @@
+        case reset:
+            return new ResetCommand();
+        case filter:
+            try {
```
You could make another subclass of Command to help handle filter commands like what you did above.

Lava-Iris replied:
Oh, I do have a subclass of Command (FilterCommand). I initiate in on line 73. Here, I'm just trying to check the input, and separate it into two cases: when we want the deadlines and events on a date and when we need all deadlines and events in a certain period (both start and end date-time is provided). 

I have different constructors for both so I need to do it here. Also, OOP-wise, I think it's the work of Parser.

Thank you for the suggestion though. Do let me know if I misunderstood your suggestion.

Merrickneo commented on src/main/java/iris/Ui.java:
```diff
@@ -0,0 +1,43 @@
+     */
+    public void greet() {
+        output(GREETING);
+    }
```
Could have a `endProgram` function to help with the closing of your `Ui` when the user inputs `bye`.

Merrickneo reviewed (commented):
Hi Lavanya, excellent use of OOP to help with your program flow and logic.
Couldn't really pinpoint many major problems and your documentation is short and descriptive as well.

Keep it up!

Lava-Iris committed:
create a simple GUI for the chatbot

Lava-Iris committed:
add assertions to check commands are working properly

Lava-Iris committed:
increase abstraction and better code quality

Lava-Iris committed:
Merge pull request #7 from Lava-Iris/branch-A-Assertions

add assertions to check code is working properly

Lava-Iris committed:
Merge pull request #8 from Lava-Iris/branch-A-CodeQuality

Increase abstraction and better code quality

Lava-Iris committed:
set up CI

Lava-Iris committed:
Merge branch 'master' of github.com:Lava-Iris/ip

Lava-Iris committed:
correct checkstyle violation in Task class

Lava-Iris committed:
Correct some outputs to match the expected output

Some tests are not passing due to change in output format.

Output format needs to consistent for better user experience.

Changing the output format of date in deadline and event to be more descriptive and consistent with the expected output.

Descriptive date in output will make it easier for user to read.

Lava-Iris committed:
close scanner object

Lava-Iris committed:
move the ui focused files to ui package

Lava-Iris committed:
change the dialogbox to allow the full message to be shown

Lava-Iris committed:
change color of messages from Iris to keep in line with lavender theme

Lava-Iris committed:
add an initial greeting when the chatbot is run

Lava-Iris committed:
add 'iris' as a title to the chatbot window

Lava-Iris committed:
close the application when 'bye' command is entered

Lava-Iris committed:
Make the GUI prettier

Lava-Iris committed:
remove batch imports using *

Lava-Iris committed:
remove unneccessary imports

Lava-Iris committed:
remove i/o tests (no longer relevant)

Lava-Iris committed:
Revert "remove i/o tests (no longer relevant)"

This reverts commit ac91a22107f8b94aade0688213a6384f3c5fff81.

Lava-Iris committed:
remove i/o tests which are no longer needed

Lava-Iris committed:
changed the output meesage for reset command

Lava-Iris committed:
add representative screenshot of product in docs

Lava-Iris committed:
change the help message

Lava-Iris committed:
change the filter logic and the output of filter command

Lava-Iris committed:
add User Guide

Lava-Iris committed:
Update README.md
